src/app/report/edit-display/edit-display.component.html

```diff
@@ -1,114 +1,118 @@
-  </div>
-
-  <div *ngIf="null === report.message" id="nullMessageLabel" class="btn btn-static my-2 mx-1 px-2">
-    Message is null
-  </div>
-
-  <div *ngIf="'' === report.message" id="emptyMessageLabel" class="btn btn-static my-2 mx-1 px-2">
-    Message is empty string
-  </div>
-
-  <div *ngIf="report.stubbed" id="stubbedMessageLabel" class="btn btn-static my-2 mx-1 px-2">
-    Message is stubbed
-  </div>
-
-  <div *ngIf="null !== report.encoding" id="encodedMessageLabel" class="btn btn-static my-2 mx-1 px-2">
-    Message of type {{report.messageClassName}} is encoded to string using {{report.encoding}}
-    <span *ngIf="this.report.showConverted">and displayed using Base64 decoded to byte array and byte array decoded to string using UTF-8 (toggle with Base64 button)</span>
-  </div>
-
-  <div *ngIf="report.preTruncatedMessageLength > 0" id="truncatedMessageLabel" class="btn btn-static my-2 mx-1 px-2">
-    Message is truncated ( {{ report.preTruncatedMessageLength - report.message.length}} characters removed)
-  </div>
-
-  <div *ngIf="null !== report.stubNotFound" id="stubNotFoundLabel" class="btn btn-static my-2 mx-1 px-2">
-    Could not find stub message for {{report.stubNotFound}}
-  </div>
-</div>
-
-<app-custom-editor [readOnlyMode]="!editingEnabled" [height]="500" data-cy-test-editor="editor" id="editor"></app-custom-editor>
-
-<div *ngIf="!editingRootNode" class="pt-4">
-  <app-display-table [report]="report"></app-display-table>
-</div>
-
-<div *ngIf="editingRootNode">
-  <div class="form-group">
-    <label>Name</label>
-    <input #name type="text" class="form-control" placeholder="Name" value="{{report.name}}">
-  </div>
-  <div class="form-group">
-    <label>Description</label>
-    <input #description class="form-control" placeholder="Description" value="{{report.description}}">
-  </div>
-  <div class="form-group">
-    <label>Path</label>
-    <input #path type="text" class="form-control" placeholder="/path/to/report/" value="{{report.path}}">
-  </div>
-  <div class="form-group">
-    <label>Transformation</label>
-    <div>
-      <textarea #transformation class="form-control transformation" >{{report.transformation}}</textarea>
+    @if (metadataTableVisible) {
+      <app-display-table [report]="report"></app-display-table>
+    }</div>
+}
+
+@if (editingEnabled) {
+  <div class="m-1" data-cy-report-edit-form>
+    <div class="form-group">
+      <label>Name</label>
+      <input #name data-cy-report-edit-form="name" type="text" class="form-control" placeholder="Name" value="{{report.name}}">
     </div>
-  </div>
-  <div class="form-group">
-    <label>Variables</label>
-    <div>
-      <textarea #variables class="form-control variables">{{report.variableCsv}}</textarea>
+    <div class="form-group">
+      <label>Description</label>
+      <input #description class="form-control" placeholder="Description" value="{{report.description}}">
```

Comment:
Attribute binding for value could be used here

Reply:
Fixed